fix(auth): Set "activeorg" when logging in via org login pages

JoshFerge opened the pull request:
### Summary
If you:
- have multiple org memberships
- do not have a url in session var _next (you navigated to the login page directly instead of being redirected),
- You can be taken to the wrong org after logging in.

This is because in get_login_redirect we set the default redirect to /auth/login,
which for an authenticated user will pull in the first org listed by Organizations.objects.get_for_user

This fix proposal sets the "activeorg" when sucessfully logging in. Other alternatives considered were changing default redirect behavior on org login pages, but this change is more constrained. 

Want to get confirmation of approach before adding onto more, there are a few more SSO cases where you can be taken to the wrong page after login. Security wise, it seems safe as in subsequent redirects we confirm org membership in `get_active_organization` before doing any org specific logic, and we check for membership when setting as well.

### TODO:
- [ ] create a diagram for login/auth redirect behaviors
- [x] write test for multi org login behavior
- [x] add activeorg to other SSO login cases
- [x] probably want to confirm superuser behavior isn't affected either locally or in staging.

JoshFerge committed:
set active org when logging in

JoshFerge committed:
dont check auth var on request

JoshFerge commented on src/sentry/auth/helper.py:
```diff
@@ -158,6 +159,8 @@ def handle_existing_identity(
         sample_rate=1.0,
     )
 
+    if request.user and not is_active_superuser(request):
```
i think checking `request.user` is redundant, we do it when we set activeorg in endpoint code but i think its always exists here.

dcramer replied:
its possible for request.user to be None technically speaking

JoshFerge replied:
(right, mean not `None`)

dcramer commented:
In general this change seems fine, however I wonder if we really need to persist that invalid default redirect? Its been a long time since I wrote/contributed/looked at this code so hard to recall what is/isnt good

JoshFerge commented:
> In general this change seems fine, however I wonder if we really need to persist that invalid default redirect? Its been a long time since I wrote/contributed/looked at this code so hard to recall what is/isnt good

I agree -- I went down that path [here](https://github.com/getsentry/sentry/pull/25609) -- it sprawled and as an auth newbie i didn't want to go around changing _a lot_ of behavior without spending a bit more time with the code / gaining more context. We redirect a lot back to `/auth/login` and route the user from there and I'm not sure if that was an architectural choice or something that was just slowly built on top of.

dcramer commented:
The one thing I would say - try to add a test to cover this. Otherwise who knows if this fixes it and that we wont insta break it again.

JoshFerge committed:
add test case for basic org login w/ multiple org memberships

JoshFerge committed:
add sso multi org redirect test case

JoshFerge committed:
handle merge case and add sso tests

JoshFerge committed:
remove request.user check, add comments

JoshFerge commented:
added 2 test cases and modified one existing one to check sso login, sso signup, and normal logins with redirects. these three tests would fail on master but pass here.

JoshFerge committed:
fix merge conflicts
